src/librustc_session/config.rs

```diff
@@ -69,6 +69,19 @@ impl FromStr for Sanitizer {
     }
 }
 
+/// The different settings that the `-Z strip` flag can have.
+#[derive(Clone, Copy, PartialEq, Hash, Debug)]
+pub enum Strip {
+    /// Do not strip at all.
+    None,
+
+    /// Strip all symbols.
+    Symbols,
+
+    /// Strip debuginfo.
+    Debuginfo,
```

Comment:
Nit: perhaps switch the order between `Symbols` and `Debuginfo` so the variant go from "least stripped" to "most stripped".